messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistSdk.kt

```diff
@@ -107,12 +115,57 @@ object GistSdk : Application.ActivityLifecycleCallbacks {
         }
     }
 
+    /**
+     * Reset the Gist SDK to its initial state.
+     * This method is used for testing purposes only.
+     */
+    internal fun reset() {
+        isInitialized = false
+        application.unregisterActivityLifecycleCallbacks(this)
+        observeUserMessagesJob?.cancel()
+        observeUserMessagesJob = null
+        gistQueue.clearUserMessagesFromLocalStore()
+        gistModalManager.clearCurrentMessage()
+        resumedActivities.clear()
+        listeners.clear()
+
+        gistQueue = Queue()
+        gistModalManager = GistModalManager()
+        currentRoute = ""
+    }
+
     fun setCurrentRoute(route: String) {
+        if (currentRoute == route) {
+            Log.i(GIST_TAG, "Current gist route is already set to: $currentRoute, ignoring new route")
+            return
+        }
+
+        cancelActiveMessage()
         currentRoute = route
         gistQueue.fetchUserMessagesFromLocalStore()
         Log.i(GIST_TAG, "Current gist route set to: $currentRoute")
     }
 
+    /**
+     * Cancels any active message being loaded if the page rule does not match the new route.
+     */
+    private fun cancelActiveMessage() {
+        val currentMessage = gistModalManager.currentMessage
+        val currentMessageProperties = currentMessage?.let { message ->
+            GistMessageProperties.getGistProperties(message)
+        }
+
+        // If there is no active message or the message does not have a route rule, we don't need to do anything
+        if (currentMessage == null || currentMessageProperties?.routeRule == null) {
```

Comment:
we need to also check for regex, in case the route rule isn't null but regex matches we don't need to cancel it.

Reply:
Good catch, updated PR